gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineExactDependencies.java

```diff
@@ -120,19 +117,6 @@ private static void configureSourceSet(
                         }
                     });
 
-                    // Without this, the 'checkUnusedDependencies correctly picks up project dependency on java-library'
-                    // test fails, by not causing gradle run the jar task, but resolving the path to the jar (rather
-                    // than to the classes directory), which then doesn't exist.
-
-                    // Specifically, we need to pick up the LIBRARY_ELEMENTS_ATTRIBUTE, which is being configured on
-                    // compileClasspath in JavaBasePlugin.defineConfigurationsForSourceSet, but we can't reference it
-                    // directly because that would require us to depend on Gradle 5.6.
-                    // Instead, we just copy the attributes from compileClasspath.
-                    compileClasspath.get().getAttributes().keySet().forEach(attribute -> {
-                        Object value = compileClasspath.get().getAttributes().getAttribute(attribute);
-                        conf.getAttributes().attribute((Attribute<Object>) attribute, value);
-                    });
```

Comment:
The reasons I think this is safe to remove is that from the comment we only want the `LIBRARY_ELEMENTS_ATTRIBUTE`, which we already set previously. The other attributes we may not actually need. If we really wanted to we could either call the internal [JvmPluginServices#configureAsCompileClasspath](https://github.com/gradle/gradle/blob/080ed571d872395e25b6e7ae3af449991016524a/subprojects/platform-jvm/src/main/java/org/gradle/api/plugins/jvm/internal/DefaultJvmPluginServices.java#L63) (dangerous imo) or copy what it's doing with the extra attributes;

Reply:
I think this makes sense, otherwise, the stuff in line 116 is wrong, and not sufficient right? since this bit of code always run unconditionally.

An aside, do you want to remove the gradle 5.6 conditional check? in line 116-119?

Reply:
Yeah, I might as well remove that.